Mage.Sets/src/mage/cards/a/ApothecaryWhite.java

```diff
@@ -0,0 +1,172 @@
+package mage.cards.a;
+
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.AttacksWithCreaturesTriggeredAbility;
+import mage.abilities.common.SimpleActivatedAbility;
+import mage.abilities.costs.Cost;
+import mage.abilities.costs.VariableCostImpl;
+import mage.abilities.costs.VariableCostType;
+import mage.abilities.costs.common.TapSourceCost;
+import mage.abilities.costs.common.TapTargetCost;
+import mage.abilities.costs.mana.ManaCostsImpl;
+import mage.abilities.dynamicvalue.DynamicValue;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.CreateTokenEffect;
+import mage.abilities.keyword.VigilanceAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.constants.SubType;
+import mage.constants.SuperType;
+import mage.filter.common.FilterControlledPermanent;
+import mage.filter.predicate.permanent.TappedPredicate;
+import mage.game.Game;
+import mage.game.combat.CombatGroup;
+import mage.game.permanent.token.FoodToken;
+import mage.game.permanent.token.HumanToken;
+import mage.players.Player;
+import mage.target.common.TargetControlledPermanent;
+import mage.util.CardUtil;
+
+import java.util.Objects;
+import java.util.UUID;
+
+/**
+ * @author Sidorovich77
+ */
+public final class ApothecaryWhite extends CardImpl {
+
+    public ApothecaryWhite(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{3}{W}");
+
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.CLERIC);
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(4);
+
+        // Vigilance
+        this.addAbility(VigilanceAbility.getInstance());
+
+        // Whenever you attack, you create a Food token for each player being attacked.
+        this.addAbility(new AttacksWithCreaturesTriggeredAbility(
+                new CreateTokenEffect(new FoodToken(), ApothecaryWhiteValue.instance)
+                        .setText("you create a Food token for each player being attacked."), 1));
+        // {W}, {T}, Tap X untapped Foods you control: Create X 1/1 white Human creature tokens.
+        Ability ability = new SimpleActivatedAbility(new ApothecaryWhiteEffect(), new ManaCostsImpl<>("{W}"));
+        ability.addCost(new TapSourceCost());
+        ability.addCost(new ApothecaryWhiteCost());
+        this.addAbility(ability);
+    }
+
+    private ApothecaryWhite(final ApothecaryWhite card) {
+        super(card);
+    }
+
+    @Override
+    public ApothecaryWhite copy() {
+        return new ApothecaryWhite(this);
+    }
+
+    enum ApothecaryWhiteValue implements DynamicValue {
```

Comment:
don't nest these as inner classes, keep them outside the card class (still in the same file)